packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -80,103 +91,31 @@ export type BlockAttributes = {
   feeRecipient?: string;
 };
 
-export enum BlockType {
-  Full = "Full",
-  Blinded = "Blinded",
-}
-export type AssembledBodyType<T extends BlockType> = T extends BlockType.Full
-  ? BeaconBlockBody
-  : BlindedBeaconBlockBody;
-export type AssembledBlockType<T extends BlockType> = T extends BlockType.Full ? BeaconBlock : BlindedBeaconBlock;
-
-export enum BlobsResultType {
-  preDeneb,
-  produced,
-  blinded,
-}
-
-export type BlobsResult =
-  | {type: BlobsResultType.preDeneb}
-  | {type: BlobsResultType.produced; contents: deneb.Contents; blockHash: RootHex}
-  | {type: BlobsResultType.blinded};
-
-export async function produceBlockBody<T extends BlockType>(
```

Comment:
not sure why you separate `produceBlockBody` to `produceBlindedBlockBody + produceExecutionBlockBody` which cause lot of changes here. Could it be the same `produceBlockBody` method but without `commonBlockBody` param?
then it can get consumed by both `builder` + `execution` flow. I prefer less code change which make it easier to review + less error
we can do refactor separately after this PR if you see the need